apps/api/src/deployment/services/deployment-reader/database-deployment-reader.service.ts

```diff
@@ -0,0 +1,180 @@
+import { Deployment, DeploymentGroup, DeploymentGroupResource } from "@akashnetwork/database/dbSchemas/akash";
+import { Op } from "sequelize";
+import { singleton } from "tsyringe";
+
+import { RestAkashDeploymentListResponse } from "@src/types/rest/akashDeploymentListResponse";
+
+export interface DatabaseDeploymentListParams {
+  owner?: string;
+  state?: "active" | "closed";
+  skip?: number;
+  limit?: number;
+  key?: string;
+  countTotal?: boolean;
+  reverse?: boolean;
+}
+
+// For the values that are not stored in the database
+export const FALLBACK_VALUE = "FALLBACK";
+
+@singleton()
+export class DatabaseDeploymentReaderService {
+  async listDeployments(params: DatabaseDeploymentListParams): Promise<RestAkashDeploymentListResponse> {
+    const { owner, state, skip = 0, limit = 100, key, countTotal = true, reverse = false } = params;
+
+    const whereClause: Record<string, unknown> = {};
+    if (owner) {
+      whereClause.owner = owner;
+    }
+    if (state === "active") {
+      whereClause.closedHeight = null;
+    } else if (state === "closed") {
+      whereClause.closedHeight = { [Op.ne]: null };
+    }
+
+    // Handle pagination similar to HTTP service
+    // If key is provided, use it as offset (simplified key-based pagination)
+    const offset = key ? parseInt(key, 10) || 0 : skip;
+
+    const queryOptions: Record<string, unknown> = {
```

Comment:
this method contains quite some repo logic. there is even a repo for deployments https://github.com/akash-network/console/blob/main/apps/api/src/deployment/repositories/deployment/deployment.repository.ts

perhaps it's non-blocking, however i don't think not following common patterns would have saved any time